ckanext/xloader/config_declaration.yaml

```diff
@@ -119,5 +119,13 @@ groups:
             to True.
         type: bool
         required: false
+      - key: ckanext.xloader.requires_validation
```

Comment:
awesome enhancement. Also just checking, can you force the load from the gui even if validation is failed?

Reply:
@duttonw haha yup! a user could click the "Upload to DataStore" button just fine. I think in our setup we just condition the button to not show. But users could still hit it with a POST request in theory. I don't think we have ever had that issue yet fortunately.

BUT, I have add in a check to the `resource_data` POST block to check for the validation stuff. And just did a flash error and return. So users can still click the button, but it won't do anything and there will be an error message.

Let me know if that is good, or if we should also hide the button? But at least this way, it will tell them that their resource has not passed validation.

Reply:
can ```ckanext.xloader.requires_validation``` be altered to 
```ckanext.xloader.validation.requiredSoWaitForReport```
True|False
Default: False (for backwards compatibility)

And can you add another option:
ckanext.xloader.validation.enforceDefaultsIfSchemaMissing
Default: True
required: false
type: bool

Then we can set this to False so that we only wait when schema is set on the resources. 

These names may be a bit too long but does describe on what they are doing. When reading the ckan.ini without these yaml declarations being available.

Reply:
I actually find the original name more intuitive.

What about `ckanext.xloader.validation.requires_report`?